### PR TITLE
Clean up: MakeLightTable()

### PR DESCRIPTION
**Pull request by kphoenix137**
Clear compiler warnings with explicit type casting and correct case for constexpr

### CODE REVIEW

File: Source/lighting.cpp [hunk 1]
```diff
@@ -351,12 +351,12 @@ void MakeLightTable()
...
-			constexpr int range = 16;
-			for (int j = 0; j < range; j++) {
-				uint8_t color = ((range - 1) << 4) / shades * (shades - i) / range * (j + 1);
+			constexpr int RANGE = 16;
+			for (int j = 0; j < RANGE; j++) {
+				uint8_t color = ((RANGE - 1) << 4) / shades * (shades - i) / RANGE * (j + 1);
 				color = 1 + (color >> 4);
-				lightTable[j + 1] = color;
-				lightTable[31 - j] = color;
+				lightTable[static_cast<uint8_t>(j + 1)] = color;
```
Comment (DakkJaniels) [inline Source/lighting.cpp:358]: couldn't you just make `j` a uint8_t to start with so you don't need to static cast it?

Comment (kphoenix137) [reply Source/lighting.cpp:358]: I will try but I think that still might have a compiler warning

**Comment by AJenbo** [comment]
What  compiler are we looking at atm? I think @obligaron fixed all the ones for the latest VS and we normally have most fixed for GCC.

**Comment by kphoenix137** [comment]
> What compiler are we looking at atm? I think @obligaron fixed all the ones for the latest VS and we normally have most fixed for GCC.
 VS 2019

File: Source/lighting.cpp (hunk 1 above)
Comment (kphoenix137) [reply Source/lighting.cpp:358]: image
Doesn't fix the issue

**Comment by AJenbo** [comment]
Maybe it's time to upgrade to VS 2022 :D

File: Source/lighting.cpp [hunk 2]
```diff
@@ -322,22 +322,22 @@ void MakeLightTable()
 {
 	// Generate 16 gradually darker translation tables for doing lighting
 	uint8_t shade = 0;
-	constexpr uint8_t black = 0;
-	constexpr uint8_t white = 255;
+	constexpr uint8_t BLACK = 0;
+	constexpr uint8_t WHITE = 255;
```
Comment (AJenbo) [inline Source/lighting.cpp:326]: Shouty case is usually reserved for preprocessor stuff

Comment (kphoenix137) [reply Source/lighting.cpp:326]: capitalize the first letter then? VS2019 is angry because they were constexpr and lowercase

**Comment by kphoenix137** [comment]
> Maybe it's time to upgrade to VS 2022 :D

I did install it awhile back and had a ton of problems so I went back

File: Source/lighting.cpp (hunk 2 above)
Comment (ephphatha) [reply Source/lighting.cpp:326]: Most of the code style rules are defined in Source/.clang-tidy, constants use `CamelCase`
https://github.com/diasurgical/devilutionX/blob/0ac30d101dc6ab0adf33772dc1595d7529a2556e/Source/.clang-tidy#L77-L81

File: Source/lighting.cpp [hunk 3]
```diff
@@ -351,12 +351,14 @@ void MakeLightTable()
...
-			constexpr int range = 16;
-			for (int j = 0; j < range; j++) {
-				uint8_t color = ((range - 1) << 4) / shades * (shades - i) / range * (j + 1);
+			constexpr int Range = 16;
+			for (int j = 0; j < Range; j++) {
+				uint8_t color = ((Range - 1) << 4) / shades * (shades - i) / Range * (j + 1);
 				color = 1 + (color >> 4);
-				lightTable[j + 1] = color;
-				lightTable[31 - j] = color;
+				auto idx = j + 1;
```
Comment (AJenbo) [inline Source/lighting.cpp:358]:
```suggestion
				int idx = j + 1;
```

File: Source/lighting.cpp [hunk 4]
```diff
@@ -322,22 +322,22 @@ void MakeLightTable()
 {
 	// Generate 16 gradually darker translation tables for doing lighting
 	uint8_t shade = 0;
-	constexpr uint8_t black = 0;
-	constexpr uint8_t white = 255;
+	constexpr uint8_t Black = 0;
+	constexpr uint8_t White = 255;
 	for (auto &lightTable : LightTables) {
```
Comment (AJenbo) [inline Source/lighting.cpp:327]:
```suggestion
	for (uint8_t &lightTable : LightTables) {
```

File: Source/lighting.cpp [hunk 5]
```diff
@@ -351,12 +351,14 @@ void MakeLightTable()
 		const auto shades = static_cast<int>(LightTables.size() - 1);
 		for (int i = 0; i < shades; i++) {
 			auto &lightTable = LightTables[i];
```
Comment (AJenbo) [inline Source/lighting.cpp:353]:
```suggestion
			uint8_t &lightTable = LightTables[i];
```

File: Source/lighting.cpp [hunk 6]
```diff
@@ -350,13 +350,15 @@ void MakeLightTable()
...
+			constexpr int Range = 16;
+			for (int j = 0; j < Range; j++) {
+				uint8_t color = ((Range - 1) << 4) / shades * (shades - i) / Range * (j + 1);
 				color = 1 + (color >> 4);
-				lightTable[j + 1] = color;
-				lightTable[31 - j] = color;
+				int idx = j + 1;
+				lightTable[idx] = color;
+				idx = 31 - j;
+				lightTable[idx] = color;
```
Comment (AJenbo) [inline Source/lighting.cpp:361]: You aren't modifying anything here, can just do it inline.
```suggestion
				lightTable[j + 1] = color;
				lightTable[31 - j] = color;
```

File: Source/lighting.cpp [hunk 7]
```diff
@@ -391,7 +393,8 @@ void MakeLightTable()
 					// Leaner falloff
 					scaled = factor * maxDarkness;
 				}
-				LightFalloffs[radius][distance] = static_cast<uint8_t>(scaled + 0.5F); // round up
+				scaled += 0.5F; // Round up
+				LightFalloffs[radius][distance] = static_cast<uint8_t>(scaled);
```
Comment (AJenbo) [inline Source/lighting.cpp:397]: Was this a problem for VC 2019?
I would prefer we don't start modifying values if we do not need to.

File: Source/lighting.cpp [hunk 8]
```diff
@@ -322,22 +322,22 @@ void MakeLightTable()
 {
 	// Generate 16 gradually darker translation tables for doing lighting
 	uint8_t shade = 0;
-	constexpr uint8_t black = 0;
-	constexpr uint8_t white = 255;
-	for (auto &lightTable : LightTables) {
+	constexpr uint8_t Black = 0;
+	constexpr uint8_t White = 255;
+	for (uint8_t &lightTable : LightTables) {
```
Comment (AJenbo) [inline Source/lighting.cpp:327]:
```suggestion
	for (auto &lightTable : LightTables) {
```

File: Source/lighting.cpp [hunk 9]
```diff
@@ -350,13 +350,15 @@ void MakeLightTable()
 		// Blood wall lighting
 		const auto shades = static_cast<int>(LightTables.size() - 1);
 		for (int i = 0; i < shades; i++) {
-			auto &lightTable = LightTables[i];
-			constexpr int range = 16;
-			for (int j = 0; j < range; j++) {
-				uint8_t color = ((range - 1) << 4) / shades * (shades - i) / range * (j + 1);
+			uint8_t &lightTable = LightTables[i];
```
Comment (AJenbo) [inline Source/lighting.cpp:353]:
```suggestion
			auto &lightTable = LightTables[i];
```

**Review by AJenbo — APPROVED** [review]

**Pull request merged**